modules/core/src/v2/coins/eth2.ts

```diff
@@ -0,0 +1,392 @@
+    }
+  }
+
+  /**
+   * Queries public block explorer to get the next ETH nonce that should be used for the given ETH address
+   * @param address
+   * @param callback
+   * @returns {*}
+   */
+  getAddressNonce(address: string, callback?: NodeCallback<number>): Bluebird<number> {
+    const self = this;
+    return co<number>(function*() {
+      // Get nonce for backup key (should be 0)
+      let nonce = 0;
+
+      const result = yield self.recoveryBlockchainExplorerQuery({
+        module: 'account',
+        action: 'txlist',
+        address,
+      });
+      const backupKeyTxList = result.result;
+      if (backupKeyTxList.length > 0) {
+        // Calculate last nonce used
+        const outgoingTxs = backupKeyTxList.filter(tx => tx.from === address);
+        nonce = outgoingTxs.length;
+      }
+      return nonce;
+    })
+      .call(this)
+      .asCallback(callback);
+  }
+
+  /**
+   * Builds a funds recovery transaction without BitGo
+   * @param params
+   * @param params.userKey {String} [encrypted] xprv
+   * @param params.backupKey {String} [encrypted] xprv or xpub if the xprv is held by a KRS provider
+   * @param params.walletPassphrase {String} used to decrypt userKey and backupKey
+   * @param params.walletContractAddress {String} the ETH address of the wallet contract
+   * @param params.recoveryDestination {String} target address to send recovered funds to
+   * @param callback
+   */
+  recover(
+    params: RecoverOptions,
+    callback?: NodeCallback<RecoveryInfo | OfflineVaultTxInfo>
+  ): Bluebird<RecoveryInfo | OfflineVaultTxInfo> {
+    throw new Error('recover not implemented');
+  }
+
+  /**
+   * Make a query to Etherscan for information such as balance, token balance, solidity calls
+   * @param query {Object} key-value pairs of parameters to append after /api
+   * @param callback
+   * @returns {Object} response from Etherscan
+   */
+  recoveryBlockchainExplorerQuery(query: any, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+    return co(function*() {
+      const response = yield request.get(common.Environments[self.bitgo.getEnv()].eth2ExplorerBaseUrl).query(query);
+
+      if (!response.ok) {
+        throw new Error('could not reach BeaconScan');
+      }
+      return response.body;
+    })
+      .call(this)
+      .asCallback(callback);
+  }
+
+  /**
+   * Fetch fee estimate information from the server
+   * @param {Object} params The params passed into the function
+   * @param {String} [params.recipient] The recipient of the transaction to estimate a send to
+   * @param {String} [params.data] The ETH tx data to estimate a send for
+   * @param callback
+   * @returns {Object} The fee info returned from the server
+   */
+  feeEstimate(params: FeeEstimateOptions, callback?: NodeCallback<FeeEstimate>): Bluebird<FeeEstimate> {
+    const self = this;
+    return co<FeeEstimate>(function* coFeeEstimate() {
```

Comment:
nit: IMO we shouldn't implement this because we really don't even know how the fees will work yet and definitely don't have an estimator implementation in the server yet